lib/Connection.php

```diff
@@ -112,6 +112,11 @@ public function pushMessage(Message $message, ?bool $masked = null): void
     // Pull a message from stream
     public function pullMessage(): Message
     {
+        // non-blocking way to wait for stream to change
+        $read = [$this->stream];
+        while (!stream_select($read, null, null, 1, 0)) {
```

Comment:
This should wait until `$this->options['timeout']`, since it's a non-blocking wait, and we can avoid the while loop.